### DISCUSSION TITLE
The share path of docker :stream path how to delete or limit growth of files

### DISCUSSION THREAD

**Discussion by nice2lee** [datalust/seq-tickets#1835] [Help]
In my seq, the files in the container mapping directory stream keep growing. Is there any way to stop or clean it up regularly?
Eg:docker run -v /data/seq:/data
in the machine /data/seq/Stream/* path there are so many files. always fills the disk
Plz help me Thanks a lot.
have a good day!
![Snipaste0606](https://user-images.githubusercontent.com/82865940/236624903-1a38e29e-530e-4542-b2d3-ee872c84eae5.png)

**Comment by KodrAus** [comment]
Hi @nice2lee 👋 

You can use [retention policies](https://docs.datalust.co/docs/retention-policies) in Seq to age out data and reclaim disk space.

Hope that helps!

**Reply by nice2lee** [reply]
Thanks a lot
 
have a  try ！

**Reply by nice2lee** [reply]
![image](https://github.com/datalust/seq-tickets/assets/82865940/41cc70fc-4542-48da-9d75-dfb6ea49d61f)
你好，我的翻译失效了，该问题还是存在，数据依然不断增长，并没有按照配置的策略及时清理以往数据，请问还有别的方式吗

**Reply by nice2lee** [reply]
From the disk, the data is still growing and has not been cleaned up in a timely manner

**Reply by nblumhardt** [reply]
Hi @nice2lee - are you able to share a screenshot of the chart and legend in Seq's _Data > Storage_ screen? Thanks!

**Reply by KodrAus** [reply]
Hi @nice2lee. Would you be happy to grab a diagnostic report through the UI by visiting _Settings > Diagnostics > Download a full diagnostic report_ and send it through to us as `support@datalust.co` and we'll take a look?

**Reply by nice2lee** [reply]
Hey Bro Thanks so much ,you are too kind!!!!!!!!!
It just filled up my disk, I emptied it once, don't know if this diagnosis is useful
[seq-diagnostics-8db5a71242bfe6e.txt](https://github.com/datalust/seq-tickets/files/11526729/seq-diagnostics-8db5a71242bfe6e.txt)

**Reply by nblumhardt** [reply]
Hi again; because the disk was emptied, the diagnostics don't contain any useful information, unfortunately.

One thing to check/think about is the storage device - Seq works with reliable block storage (e.g. local disks, Azure Managed Disks, EBS) but not generally with network filesystems (many of these don't support flushing properly, don't support file locks, etc.). If you're not using a regular disk, this could be at the root of the issue.

Otherwise, if the problem appears again, capturing a screenshot from the chart at the top of Data > Storage should cast some light on it.

Hope this helps,
Nick

**Reply by nice2lee** [reply]
Hi again;The next time a similar problem occurs, the log will be sent to you，Thanks a lot!!!
![Snipaste2312](https://github.com/datalust/seq-tickets/assets/82865940/08b47c7e-5b84-48f9-b281-53ead8aea56c)

**Reply by nice2lee** [reply]
hi again;
![Snipaste2636](https://github.com/datalust/seq-tickets/assets/82865940/79840a1b-d887-4636-99f6-0ac5912b61a9)
[seq-diagnostics-8db5d9fcaf73fb8.txt](https://github.com/datalust/seq-tickets/files/11571482/seq-diagnostics-8db5d9fcaf73fb8.txt)
Plz help.

**Reply by liammclennan** [reply]
Hi @nice2lee 

The oldest events have not yet hit the 5 day retention policy. If you want them to be deleted sooner you will need an earlier retention policy.

**Reply by nice2lee** [reply]
Hi @liammclennan  Then I will continue to observe, thank you！

**Reply by nice2lee** [reply]
Hi，Iam here again.Iam so sorry~The configuration shows that all events are cleared after 3 days. But the 'Stream' content in the file does not affect each other, the current file in the 'Stream' directory is more than 3 days, do I still lack any configuration?
![Snipaste2913](https://github.com/datalust/seq-tickets/assets/82865940/3bc56df9-a387-4e88-98fe-79877eb7180f)
![Snipaste2937](https://github.com/datalust/seq-tickets/assets/82865940/c03f9d50-da13-42cb-aae9-4ccd088cf9d6)

**Reply by nblumhardt** [reply]
Hi Lee; these are all .junk files, which Seq won't clean up at all. They're left behind when Seq detects that there's been some (recoverable) storage corruption; they'll contain log data that Seq found "lying around" in the filesystem that wasn't part of the current file set.

You can safely delete anything with the .junk extension. However, the fact that they're here suggests that whatever is being used for storage isn't suitable for Seq (is it a network drive, for example?).

Hope this helps,
Nick

**Reply by nice2lee** [reply]
Hi  Nick，Then I'll clean up these files regularly.
Thanks a lot!!! hava a good day!
